streaming/base/stream.py

```diff
@@ -400,8 +421,7 @@ def get_shards(self, world: World) -> List[Reader]:
         filename = os.path.join(self.local, self.split, basename)  # pyright: ignore
         if world.is_local_leader:
             if self.remote:
-                tmp_filename = self._download_file(basename, basename + '.tmp')
-                os.rename(tmp_filename, filename)
+                filename = self._download_file(basename)
```

Comment:
the following changes cause the following error on a single node, multi-gpu setup(4 gpus):
```
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Index file at /tmp/local_path_1/train/index.json is empty or corrupted. Expecting value: line 1 column 1 (char 0)
```

On re -running the script the issue is fixed. But every time I change the `local` path, it always gives this error in the first run and fixes itself from the second run. I tried running with the earlier code and the code runs fine every time. What could be the reason? Why was this here in the first place? I mean the downloading and renaming part.